BridgeSDK/BridgeAPI/SBBActivityManager.m

```diff
@@ -56,13 +58,80 @@ + (instancetype)defaultComponent
     return shared;
 }
 
-- (NSURLSessionDataTask *)getScheduledActivitiesForDaysAhead:(NSInteger)daysAhead withCompletion:(SBBActivityManagerGetCompletionBlock)completion
+- (NSURLSessionDataTask *)getScheduledActivitiesForDaysAhead:(NSInteger)daysAhead withCompletion:(SBBActivityManagerGetCompletionBlock)completion {
+    SBBCachingPolicy policy = gSBBUseCache ? SBBCachingPolicyFallBackToCached : SBBCachingPolicyNoCaching;
+    return [self getScheduledActivitiesForDaysAhead:daysAhead cachingPolicy:policy withCompletion:completion];
+}
+
+- (NSURLSessionDataTask *)getScheduledActivitiesForDaysAhead:(NSInteger)daysAhead cachingPolicy:(SBBCachingPolicy)policy withCompletion:(SBBActivityManagerGetCompletionBlock)completion
+{
+    return [self getScheduledActivitiesForDaysAhead:daysAhead daysBehind:0 cachingPolicy:policy withCompletion:completion];
+}
+
+- (NSURLSessionDataTask *)getScheduledActivitiesForDaysAhead:(NSInteger)daysAhead daysBehind:(NSInteger)daysBehind cachingPolicy:(SBBCachingPolicy)policy withCompletion:(SBBActivityManagerGetCompletionBlock)completion
 {
     NSMutableDictionary *headers = [NSMutableDictionary dictionary];
     [self.authManager addAuthHeaderToHeaders:headers];
     
+    __block SBBResourceList *tasks = nil;
+    NSArray *daysBehindTasks = [NSArray array];
+    id<SBBCacheManagerProtocol> cacheManager = nil;
+    if (policy != SBBCachingPolicyNoCaching) {
+        if ([self.objectManager conformsToProtocol:@protocol(SBBObjectManagerInternalProtocol)]) {
+            cacheManager = ((id<SBBObjectManagerInternalProtocol>)self.objectManager).cacheManager;
+        }
+        tasks = (SBBResourceList *)[cacheManager cachedSingletonObjectOfType:@"ResourceList" createIfMissing:NO];
+        
+        // if we're going straight to cache, we're done
+        if (policy == SBBCachingPolicyCachedOnly) {
+            dispatch_async(dispatch_get_main_queue(), ^{
+                if (completion) {
+                    completion(tasks, nil);
+                }
+            });
+            return nil;
+        }
+        
+        // otherwise, keep unfinished tasks from yesterday so we can show them in the Yesterday section
+        // (because the server doesn't give us those)
+        if (daysBehind > 0) {
```

Comment:
Seems to me that you'd want the app developers to be able to filter on that rather than assume they only want the unfinished ones.


Sent from my iPhone

> On Apr 25, 2016, at 5:10 PM, Erin-Mounts <notifications@github.com> wrote:
> 
> In BridgeSDK/BridgeAPI/SBBActivityManager.m:
> 
> > +        }
> > +        tasks = (SBBResourceList *)[cacheManager cachedSingletonObjectOfType:@"ResourceList" createIfMissing:NO];
> > +        
> > +        // if we're going straight to cache, we're done
> > +        if (policy == SBBCachingPolicyCachedOnly) {
> > +            dispatch_async(dispatch_get_main_queue(), ^{
> > +                if (completion) {
> > +                    completion(tasks, nil);
> > +                }
> > +            });
> > +            return nil;
> > +        }
> > +        
> > +        // otherwise, keep unfinished tasks from yesterday so we can show them in the Yesterday section
> > +        // (because the server doesn't give us those)
> > +        if (daysBehind > 0) {
> ah, yes, good point... but only the ones finished today, right?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
>